### PR TITLE
hubble: allow filtering by agent event subtypes

### PR DESCRIPTION
**Pull request by tklauser**
Follow-up for #14242

The first commit updates the protobuf enum values for the agent event types, so they match the respective monitor API values. This will make the implementation on the Hubble CLI side more straight-forward.

The second commit introduces functionality to filter agent events by event type (e.g. endpoint creation, policy update).

### CODE REVIEW

**Comment by tklauser** [comment]
test-me-please

File: api/v1/flow/flow.proto [hunk 1]
```diff
@@ -477,17 +477,20 @@ message LostEvent {
...
-    ENDPOINT_REGENERATE_FAILURE = 5;
-    ENDPOINT_CREATED = 6;
-    ENDPOINT_DELETED = 7;
-    IPCACHE_UPSERTED = 8;
-    IPCACHE_DELETED = 9;
-    SERVICE_UPSERTED = 10;
-    SERVICE_DELETED = 11;
+    // used for AGENT_EVENT_GENERIC in monitor API, but there are currently no
+    // such events;
+    reserved 1;
```
Comment (rolinh) [inline api/v1/flow/flow.proto:482]: Generally speaking, we can't reassign values. I understand this breaking change is OK here as `AgentEventType` is not in any stable Cilium release.

**Review by rolinh — APPROVED** [review]
I guess we still need to have an API discussion to know if https://github.com/cilium/cilium/pull/14242#issuecomment-737831025 also needs to be tackled.

File: api/v1/flow/flow.proto (hunk 1 above)
Comment (tklauser) [reply api/v1/flow/flow.proto:482]: Indeed, I should have mentioned this is a breaking change in the commit message at least. As you pointed out it is not part of any stable release, so I figure it should be fine.

**Comment by tklauser** [comment]
retest-runtime

**Comment by tklauser** [comment]
retest-net-next

**Review by glibsm — APPROVED** [review]

**Comment by nebril** [comment]
retest-1.19-4.19

**Comment by nebril** [comment]
retest-4.19

**Comment by nebril** [comment]
retest-4.9

**Pull request merged**